django/forms/formsets.py

```diff
@@ -325,8 +325,12 @@ def full_clean(self):
             # Empty forms are unchanged forms beyond those with initial data.
             if not form.has_changed() and i >= self.initial_form_count():
                 empty_forms_count += 1
-
-            self._errors.append(form.errors)
+            # Call form.errors first so we can get cleaned_data.
+            form_errors = form.errors
+            if self.can_delete:
+                if self._should_delete_form(form):
+                    continue
```

Comment:
You can combine the `_should_delete_form` check using `self.can_delete and ...` above.